src/nylas-connection.ts

```diff
@@ -101,7 +108,8 @@ export default class NylasConnection {
   }
 
   requestOptions(options: RequestOptions): RequestOptions {
-    const url = new URL(`${config.apiServer}${options.path}`);
+    const baseUrl = this.baseUrl ? this.baseUrl : config.apiServer;
+    const url = new URL(`${baseUrl}${options.path}`);
```

Comment:
Implemented a way to override the base URL for outgoing requests in a `NylasConnection` object.